### PR TITLE
Breadcrumbs docs

### LINKED ISSUE #249: Add documentation for breadcrumbs
We currently have no documentation for the breadcrumbs feature, so this will need to be added before launch.

@matsimitsu Just checking to ensure I'm understanding how the fields should be used:

`action` = UI | Network | Navigation | Console
`message` = “The user clicked a button” | “HTTP 500 from http://blablabla.com” | “The user pressed back” | “A console.log message”

`metadata` = key/value metadata in <string, string> format

Does this seem okay to you?

### PR DESCRIPTION
**Pull request by xadamy**
Adds documentation for the breadcrumbs feature. Closes #249.

### CODE REVIEW

File: source/front-end/breadcrumbs.html.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Appsignal from "@appsignal/javascript"
+
+const appsignal = new Appsignal({ 
+  key: "YOUR FRONTEND API KEY"
+})
+
+appsignal.addBreadcrumb({
+  category: "", // e.g. "UI", "Network", "Navigation", "Console"
+  action: "", // e.g "The user clicked a button", "HTTP 500 from http://blablabla.com"
+  metadata: {} // key/value metadata in <string, string> format
```
Comment (matsimitsu) [inline source/front-end/breadcrumbs.html.md]: maybe add the `message` param as well?

Comment (xadamy) [reply source/front-end/breadcrumbs.html.md]: @matsimitsu I thought we had dropped this in favour of the `action` or `category` field? Can you confirm?

Comment (matsimitsu) [reply source/front-end/breadcrumbs.html.md]: No message is still there, you can use either/and message and metadata.Message is for long stuff and metadata for key/value things (or at least that's how it's implemented in the view):)

Comment (xadamy) [reply source/front-end/breadcrumbs.html.md]: > you can use either/and message and metadata

So you mean you can use either `message` or `metadata`, and both at the same time?

Comment (matsimitsu) [reply source/front-end/breadcrumbs.html.md]: Correct!

Comment (xadamy) [reply source/front-end/breadcrumbs.html.md]: Thanks! 👍

File: source/front-end/breadcrumbs.html.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+|  metadata  |  Object<string, string>  |  (optional) An object of metadata related to the event  |
+
+## Plugins
+
+There are a number of [plugins](#plugins) available to enable automatic collection of breadcrumbs, and more will be added as time goes on:
+
+- [`@appsignal/plugin-breadcrumbs-console`](https://github.com/appsignal/appsignal-javascript/tree/develop/packages/plugin-breadcrumbs-console)
+- [`@appsignal/plugin-breadcrumbs-network`](https://github.com/appsignal/appsignal-javascript/tree/develop/packages/plugin-breadcrumbs-network)
+
+Have a good idea for a breadcrumbs plugin?
```
Comment (matsimitsu) [inline source/front-end/breadcrumbs.html.md]: I feel like this sentence isn't finished yet :)

File: source/front-end/breadcrumbs.html.md [hunk 3]
```diff
@@ -29,9 +33,18 @@ appsignal.addBreadcrumb({
...
+| ------ | ------ | ----- |
+|  category  |  String  |  Category to label the event under  |
+|  action  |  String  |  Contextual information related to the event  |
+|  message  |  String  |  (optional) A log message or other string to send to AppSignal  |
+|  metadata  |  Object<string, string>  |  (optional) An object of metadata related to the event  |
+
 ## Plugins
 
-There are a number of plugins available to enable automatic collection of breadcrumbs, and more will be added as time goes on:
+There are a number of [plugins](#plugins) available to enable automatic collection of breadcrumbs, and more will be added as time goes on:
```
Comment (wesoudshoorn) [inline source/front-end/breadcrumbs.html.md]: All our headings get automagic links, so the heading above should already be linkable through `#plugins`. Think we can remove this inline anchor?

File: source/front-end/breadcrumbs.html.md (hunk 2 above)
Comment (xadamy) [reply source/front-end/breadcrumbs.html.md]: Indeed, fixed!

**Pull request merged**